src/path_rewriting.rs

```diff
@@ -91,14 +91,20 @@ fn apply_mapping(mapping: &Option<Value>, path: &str) -> PathBuf {
     PathBuf::from(path)
 }
 
-// Remove common part between the prefix's end and the path's start
+// If the join of the source and the relative path is a file, return it.
+// Otherwise, remove common part between the source's end and the relative
+// path's start.
 fn guess_abs_path(prefix_dir: &PathBuf, path: &PathBuf) -> PathBuf {
+    let full_path = prefix_dir.join(path);
+    if full_path.is_file() {
+        return full_path;
+    }
     for ancestor in path.ancestors() {
         if prefix_dir.ends_with(ancestor) && !ancestor.as_os_str().is_empty() {
             return prefix_dir.join(path.strip_prefix(ancestor).unwrap().to_path_buf());
         }
     }
-    prefix_dir.join(path)
+    full_path
 }
```

Comment:
I was thinking of an alternative solution to avoid any I/O (`is_file` does I/O).

Instead of stopping at the first ancestor for which `prefix_dir.ends_with(ancestor)`, what if we stop at the last one?

Reply:
I don't fully grasp the purpose of stripping here, but I don't think that that would work in the case this PR tries to address. For example,  in the test case I added, the source directory `S`  and the relative path `P` have stuff in common, but what we want is really `S + P`, i.e. no stripping.

Reply:
To try to understand `guess_abs_path`, I've changed its implementation to simply be `prefix_dir.join(path)` (so, no stripping) and the tests seem to pass. @marco-c  Could it be that we don't need to perform any stripping here?

Reply:
Right, I see no other way.

BTW, the `guess_abs_path` function was introduced in #197. I don't recall exactly what was the real world problem it fixed.

Reply:
> To try to understand guess_abs_path, I've changed its implementation to simply be prefix_dir.join(path) (so, no stripping) and the tests seem to pass. @marco-c Could it be that we don't need to perform any stripping here?

Interesting, I thought the test added as part of #197 would fail without that "guess"!

Reply:
From what I understand, the added test is not testing the guessing part. It seems that that test passes due to the `assert` in the added 
```rust
    if let Some(p) = &source_dir {
        assert!(p.is_absolute());
    }
```
failing.